lg69t/firmware_tool.py

```diff
@@ -226,9 +227,41 @@ def print_bytes(byte_data):
         [f'0x{c:02X}' for c in byte_data]
     ))
 
+def extract_fw_files(p1fw):
+    app_bin_fd = None
+    gnss_bin_fd = None
+    if isinstance(p1fw, ZipFile):
+        for filename in p1fw.namelist():
+            if filename.endswith('upg.bin'):
```

Comment:
We didn't have any criteria in the past for attempting to use a file as firmware except "the user told me so". Now, we do have a criterion: the file name ends in `upg.bin` or `sta.bin`. I suppose that's basically the same -- "the user told me to use this directory or zip file". The difference is now that this code requires that _we_ always name our firmware files in accordance with this undocumented assumption. If something else in our system guarantees that these files are always named consistently, then no problem. Otherwise, this could bite us in the future. Is there anything that guarantees this interface between the firmware files and this code?

Reply:
I guess the only other things in the system that care about this are the release script and Eos.

The release script adds the `upg.bin` to our firmware file, but the `sta.bin` is just part of the file uploaded to S3.

The way that Eos loads the files is that it looks at the `info.json` in the zip to decide which file is which. We should probably do this as well. The file as the form:

```
{
              "build_time": "${BUILD_DATE_ISO}",
              "device_type": "${PRODUCT}",
              "device_variant": "",
              "fusion_engine": {
                  "filename": "${FUSION_ENGINE_FILENAME}",
                  "version": "${VERSION_STR}"
              },
              "gnss_receiver": {
                  "filename": "${TESEO_FILENAME}",
                  "version": "${TESEO_VERSION}"
              },
              "boot": {
                  "filename": "${BOOTLOADER_FILENAME}",
                  "version": "${BOOTLOADER_VERSION}"
              }
          }
 ```

Reply:
Just pushed a change that extracts the filenames from `info.json`.